### ISSUE TITLE
IIS services not installed or starting on Mac OS X

### ISSUE THREAD

**Issue by jostromzyston** [rapid7/metasploitable3#168]
## Issue Description
The IIS services (tcp/80, tcp/21) are not installed or starting.  It looks like the steps aren't running from 'setup_iis.bat'.  I've been trying to manually run the commands but no luck.  Any ideas on troubleshooting or fixing this issue?  I started with a newer version of vagrant then downgraded to 1.9.1.  I have done vagrant easy ("MS3_DIFFICULTY=easy vagrant up") order to disable the firewall, but no luck.  I don't even see the world wide web service under services snapin.

## Host System
* OS:  Mac OS X Sierra (10.12.5)
* Packer Version:  1.0.2
* Vagrant Version:  1.9.1
* VirtualBox Version:  5.1.22

## nmap services scanStarting Nmap 7.40 ( https://nmap.org ) at 2017-07-19 10:16 EDT
Nmap scan report for 10.0.2.4
Host is up (0.00020s latency).
Not shown: 980 closed ports
PORT      STATE SERVICE              VERSION
22/tcp    open  ssh                  OpenSSH 7.1 (protocol 2.0)
135/tcp   open  msrpc                Microsoft Windows RPC
139/tcp   open  netbios-ssn          Microsoft Windows netbios-ssn
445/tcp   open  microsoft-ds         Microsoft Windows Server 2008 R2 - 2012 microsoft-ds
3306/tcp  open  mysql                MySQL 5.5.20-log
3389/tcp  open  tcpwrapped
4848/tcp  open  ssl/http             Oracle Glassfish Application Server
7676/tcp  open  java-message-service Java Message Service 301
8009/tcp  open  ajp13                Apache Jserv (Protocol v1.3)
8022/tcp  open  http                 Apache Tomcat/Coyote JSP engine 1.1
8031/tcp  open  ssl/unknown
8080/tcp  open  http                 Sun GlassFish Open Source Edition  4.0
8181/tcp  open  ssl/intermapper?
8443/tcp  open  ssl/https-alt?
9200/tcp  open  http                 Elasticsearch REST API 1.1.1 (name: Asmodeus; Lucene 4.7)
49152/tcp open  msrpc                Microsoft Windows RPC
49153/tcp open  msrpc                Microsoft Windows RPC
49154/tcp open  msrpc                Microsoft Windows RPC
49155/tcp open  unknown
49156/tcp open  msrpc                Microsoft Windows RPC

**Comment by Chan9390** [comment]
Hey @jostromzyston , it is a known issue. A fix will be made available soon.

**Comment by jostromzyston** [comment]
Thanks for the quick response.  Should we add this to the General Section under Mac?  I looked first under General Section to see if there were any known issues.  Quick question, if you don't mind:
1.  Is this only impacting Mac OS X?  If so I can run through this on Windows or Linux.
2. Any idea on estimated time of fix?  
Thanks a lot.

**Comment by jostromzyston** [comment]
@Chan9390
Thanks for the quick response. Should we add this to the General Section under Mac? I looked first under General Section to see if there were any known issues. Quick question, if you don't mind:

Is this only impacting Mac OS X? If so I can run through this on Windows or Linux.
Any idea on estimated time of fix?

**Comment by Chan9390** [comment]
**Is this only impacting Mac OS X?**
This is not specific to Mac OS because there is no problem with the virtualization software (VirtualBox). The issue is with the configuration, sometimes the service starts up and sometimes it doesn't.

**Any idea on estimated time of fix?**
IMHO a quick fix would take somewhere between 3 - 5 days. Currently we are working on converting entire process of building metasploitable 3 to a more reliable one by adding provisioners. This makes the setup stable and easy to setup. This conversion would take atleast a month to be stable itself. Also a **pre-built metasploitable 3 would be made available soon**, all you got to do is just *`vagrant init metasploitable3 && vagrant up`*.

**Comment by jostromzyston** [comment]
@Chan9390 Thank you, sir.  I look forward to seeing this pre-built metaploitable 3.

**Comment by jostromzyston** [comment]
Hey @Chan9390 I was just checking in to see if the pre-built metasploitable 3 is available yet?  I've been checking the code every morning, and saw some changes about 4 days ago.  I downloaded a new master zip but wasn't able to have any success by running `vagrant init metasploitable3 && vagrant up`

**Comment by Chan9390** [comment]
@jostromzyston The pre-built box is not yet uploaded. Will be uploaded and announced soon.

**Comment by Chan9390** [comment]
Hey @jostromzyston , pre-built Metasploitable 3 box is up on Vagrant cloud.
All you got to do is `vagrant init -m jbarnett-r7/metasploitable3-win2k8 && vagrant up`.

**Comment by jostromzyston** [comment]
Thank you @Chan9390!  I will test this out and I'm sure it should work great.

**Comment by jostromzyston** [comment]
Hi @Chan9390, the process was very smooth for bringing up the pre-built box.  One issue I noticed is that the IIS service is still not installing.  I was able to get this working before by just running the setup scripts manually.  Not sure what the situation is with this.

**Comment by Chan9390** [comment]
Okay. Will take a look at it.

**Comment by TuanThai** [comment]
I got the same issue.  I have done the build many times on both Windows an Ubuntu. However,  IIS and some other services were not in the box.  Any ideas to fix the issue? Thank you!

**Comment by spiralout311** [comment]
Ports 21 and 80 were closed for me too. I followed these instructions to set up IIS:
https://docs.microsoft.com/en-us/iis/install/installing-iis-7/installing-iis-7-and-above-on-windows-server-2008-or-windows-server-2008-r2

And this one to get ftp going:
https://docs.microsoft.com/en-us/iis/publish/using-the-ftp-service/configuring-ftp-firewall-settings-in-iis-7

I also had to disable AVG Firewall b/c it was flagging jenkins and a few others, and had to downgrade vagrant to 1.9.1. Took me 3 days to figure all that out, but was well worth it!
